pkgs/development/libraries/open62541/default.nix

```diff
@@ -0,0 +1,161 @@
+  pname = "open62541";
+  version = "1.3.5";
+
+  src = fetchFromGitHub {
+    owner = "open62541";
+    repo = finalAttrs.pname;
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-X0kdMKSqKAJvmrd1YcYe1mJpFONqPCALA09xwd6o7BQ=";
+    fetchSubmodules = true;
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "Ensure-absolute-paths-in-pkg-config-file.patch";
+      url = "https://github.com/open62541/open62541/pull/5150.patch";
+      sha256 = "sha256-mq4h32js2RjI0Ljown/01SXA3gc+7+zX8meIcvDPvoA=";
+    })
+  ];
+
+  cmakeFlags = [
+    "-DOPEN62541_VERSION=v${finalAttrs.version}"
+
+    "-DBUILD_SHARED_LIBS=${if stdenv.hostPlatform.isStatic then "OFF" else "ON"}"
+    "-DUA_NAMESPACE_ZERO=FULL"
+
+    "-DUA_BUILD_UNIT_TESTS=${if finalAttrs.doCheck then "ON" else "OFF"}"
+  ]
+  ++ lib.optional withExamples "-DUA_BUILD_EXAMPLES=ON"
+  ++ lib.optional (withEncryption != false)
+    "-DUA_ENABLE_ENCRYPTION=${lib.toUpper encryptionBackend.pname}"
+  ++ lib.optional withPubSub "-DUA_ENABLE_PUBSUB=ON"
+  ;
+
+  nativeBuildInputs = with python3Packages; [
+    cmake
+    pkg-config
+    python
+  ]
+  ++ lib.optionals withDoc [
+    sphinx
+    sphinx_rtd_theme
+    graphviz-nox
+  ];
+
+  buildInputs = lib.optional (withEncryption != false) encryptionBackend;
+
+  buildFlags = [ "all" ] ++ lib.optional withDoc "doc";
+
+  doCheck = true;
+
+  checkInputs = [
+    check
+    subunit
+  ];
+
+  # Tests must run sequentially to avoid port collisions on localhost
+  enableParallelChecking = false;
+
+  preCheck = let
+    disabledTests =
+      lib.optionals (withEncryption == "mbedtls") [
+        "encryption_basic128rsa15"
+      ]
+      ++ lib.optionals withPubSub [
+        # "Cannot set socket option IP_ADD_MEMBERSHIP"
+        "pubsub_publish"
+        "check_pubsub_get_state"
+        "check_pubsub_publish_rt_levels"
+        "check_pubsub_subscribe_config_freeze"
+        "check_pubsub_subscribe_rt_levels"
+        "check_pubsub_multiple_subscribe_rt_levels"
+      ];
+    regex = "^(${builtins.concatStringsSep "|" disabledTests})\$";
+  in lib.optionalString (disabledTests != []) ''
+    checkFlagsArray+=(ARGS="-E ${lib.escapeRegex regex}")
+  '';
+
+  postInstall = lib.optionalString withDoc ''
+    # excluded files, see doc/CMakeLists.txt
+    rm -r doc/{_sources/,CMakeFiles/,cmake_install.cmake}
+
+    # doc is not installed automatically
+    mkdir -p $out/share/doc/${finalAttrs.pname}
+    cp -r doc/ $out/share/doc/${finalAttrs.pname}/html
+  '' + lib.optionalString withExamples ''
+    # install sources of examples
+    mkdir -p $out/share/${finalAttrs.pname}
+    cp -r ../examples $out/share/${finalAttrs.pname}
+
+    ${lib.optionalString (!stdenv.hostPlatform.isWindows) ''
+    # remove .exe suffix
+    mv -v $out/bin/ua_server_ctt.exe $out/bin/ua_server_ctt
+    ''}
+
+    # remove duplicate libraries in build/bin/, which cause forbidden
+    # references to /build/ in ua_server_ctt
+    rm -r bin/lib${finalAttrs.pname}*
```

Comment:
pname should not be taken from finalAttrs because when ammending suffixes to it, it is most likely wrong here